examples/empirical_frechet_mean_uncertainty_s3.py

```diff
@@ -0,0 +1,184 @@
+
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+import geomstats.backend as gs
+# import geomstats.visualization as visualization
+
+from geomstats.geometry.hypersphere import Hypersphere
+
+# n = number of samples for computing the mean
+# NN = number of trials for the stochastic integral of expectation
+# Ntheta = number of points on the curve sig_est(theta)
+
+def my_random_uniform_sphere(n):
+    #locals(x,xn)
+    # Random coordinates between -1 and 1
+    x = (np.random.random_sample(n+1)*2.0 -1.0)
+    nx = np.linalg.norm(x)
+    while nx > 1 or nx < 1.e-10:
+       x = (np.random.random_sample(n+1)*2.0 -1.0)
+       nx = np.linalg.norm(x)
+    return(x/nx)
+
+def my_random_sample(N, dim):
+    X = np.zeros( (N, dim+1), dtype=gs.float64)
+    for i in range(N):
+        X[i,:] = my_random_uniform_sphere(dim)
+    return X
+
+
+
+def empirical_var(n, theta, dim, NN=1000):
+    """
+    Compute the variance of the empirical Fréchet mean of n samples
+    drawn from an isotropic distributions on hyper sub-spheres of radius
+    0 < theta < Pi around the north pole of the sphere of dimension dim.
+    This is repeated NN times to make a stochastic approximation of the
+    expectation.
+    """
+    assert dim > 1, "Dim > 1 needed to draw a uniform sample on subsphere"
+    var = []
+    sphere = Hypersphere(dimension=dim)
+    subsphere = Hypersphere(dimension=dim-1)
+
+    # Define north pole
+    north_pole = np.zeros(dim+1)
+    north_pole[dim] = 1.0
+    for j in range(NN):
+        # Sample n points from the uniform distrib on a subsphere
+        # of radius theta (i.e cos(theta) in ambiant space)
+        data = gs.zeros((n, dim+1), dtype=gs.float64)
+        # For sampling on a subsphere, use RandomUniform(dim-1)
+        directions = subsphere.random_uniform(n)
+        # directions = my_random_sample(n, dim-1)
+        for i in range(n):
+            for j in range(dim):
+                data[i,j] = gs.sin(theta) * directions[i,j]
+            data[i,dim] = gs.cos(theta)
+        ## Compute empirical Frechet mean of the n-sample
+        current_mean = sphere.metric.adaptive_gradientdescent_mean(data, n_max_iterations=64, init_points=[north_pole])
+        var.append(sphere.metric.squared_dist(north_pole, current_mean))
+    return (np.mean(var), 2* np.std(var) / np.sqrt( NN ) )
+
+
+def modulation_factor(n, theta, dim, NN=1000):
+    """
+    Compute the modulation factor as the ratio between the variance of
+    the empirical Fréchet mean of n samples drawn from an isotropic
+    distributions on hyper sub-spheres of radius 0 < theta < Pi around
+    the north pole of the sphere of dimension dim.
+    This is repeated NN times to make a stochastic approximation of the
+    expectation and stddev on the modulation factor is returned as second parameter
+    """
+    (var, stdvar) = empirical_var(n, theta, dim, NN=NN)
+    return var * n / theta ** 2, stdvar * n / theta ** 2
+
+
+
+def asymptotic_modulation(dim, theta):
```

Comment:
A docstring + reference would be nice here (or in the docstring at the top of the file)